### PR TITLE
Implements integration test

### PR DESCRIPTION
**Pull request by stasimus**

### CODE REVIEW

File: bakery/baker-client/src/main/scala/com/ing/bakery/common/FailoverUtils.scala [hunk 1]
```diff
@@ -26,25 +25,25 @@ object FailoverUtils extends LazyLogging{
     * @return
     */
   def calWithFailOver(fos: FailoverState,
```
Comment (jacum) [inline bakery/baker-client/src/main/scala/com/ing/bakery/common/FailoverUtils.scala:27]: call (double L)?

File: bakery/baker-client/src/main/scala/com/ing/bakery/common/FailoverUtils.scala [hunk 2]
```diff
@@ -26,25 +25,25 @@ object FailoverUtils extends LazyLogging{
...
+        .expectOr(func(uri).map({ request: Request[IO] =>
+          filters.foldLeft(request)((acc, filter) => filter(acc))
+        }))(handleHttpErrors)(jsonOf[IO, BakerResult])
 
     retryingOnAllErrors(
       policy = limitRetries[IO](fos.size * 2) |+| exponentialBackoff[IO](5.milliseconds),
-      onError = (error: Throwable, _: RetryDetails) => IO {
-        logger.warn(s"Failed to call host ${fos.host}", error)
+      onError = (error: Throwable, retryDetails: RetryDetails) => IO {
+        logger.warn(s"Failed to call host ${fos.host}, retry #${retryDetails.retriesSoFar}", error)
```
Comment (jacum) [inline bakery/baker-client/src/main/scala/com/ing/bakery/common/FailoverUtils.scala:44]: will the error be logged here with full stacktrace? maybe we don't need it - just `error.getMessage` ?

File: bakery/baker-client/src/test/scala/com/ing/bakery/client/BakerClientSpec.scala [hunk 3]
```diff
@@ -97,6 +97,23 @@ class BakerClientSpec extends BakeryFunSpec {
...
+
+      BakerClient.resourceBalanced(
+        hosts = List(uri3, uri2, uri1),
+        pool = executionContext,
+        filters = List.empty,
+        tlsConfig = Some(clientTLSConfig))
+        .use { client =>
+          for {
+            response <- IO.fromFuture(IO(client.getAllRecipeInstancesMetadata))
+          } yield assert(response.isEmpty)
```
Comment (jacum) [inline bakery/baker-client/src/test/scala/com/ing/bakery/client/BakerClientSpec.scala:113]: what are we asserting exactly by `response.isEmpty`? 
how does it show that there were retries and one request was successful?

File: bakery/baker-client/src/main/scala/com/ing/bakery/common/FailoverUtils.scala (hunk 2 above)
Comment (stasimus) [reply bakery/baker-client/src/main/scala/com/ing/bakery/common/FailoverUtils.scala]: Agree, decreased the logs pollution.

**Review by jacum — APPROVED** [review]

**Pull request merged**